tox.ini

```diff
@@ -50,50 +66,45 @@ setenv =
     cdh: HADOOP_HOME={toxinidir}/.tox/hadoop-cdh
     hdp: HADOOP_DISTRO=hdp
     hdp: HADOOP_HOME={toxinidir}/.tox/hadoop-hdp
-    contrib: NOSE_ATTR=contrib
-    apache: NOSE_ATTR=apache
-    aws: NOSE_ATTR=aws
-    postgres: NOSE_ATTR=postgres
-    scheduler: NOSE_ATTR=scheduler
-    cdh,hdp: NOSE_ATTR=minicluster
-    gcloud: NOSE_ATTR=gcloud
-    unixsocket: NOSE_ATTR=unixsocket
-    dropbox: NOSE_ATTR=dropbox
-    azureblob: NOSE_ATTR=azureblob
-    core: NOSE_EVAL_ATTR=not minicluster and not gcloud and not postgres and not unixsocket and not contrib and not apache and not aws and not azureblob and not dropbox
     LUIGI_CONFIG_PATH={toxinidir}/test/testconfig/luigi.cfg
     COVERAGE_PROCESS_START={toxinidir}/.coveragerc
     FULL_COVERAGE=true
-    core: NOSE_WITH_DOCTEST=1
     AWS_DEFAULT_REGION=us-east-1
     AWS_ACCESS_KEY_ID=accesskey
     AWS_SECRET_ACCESS_KEY=secretkey
 commands =
     cdh,hdp: {toxinidir}/scripts/ci/setup_hadoop_env.sh
     azureblob: {toxinidir}/scripts/ci/install_start_azurite.sh {toxinidir}/scripts/ci
     python --version
-    coverage run test/runtests.py -v --ignore-files='(^\.|^_|^setup\.py$)' \
-    {posargs:}
-    coverage combine
+    ## main commands
+    contrib: python test/runtests.py -m contrib {posargs:}
+    apache: python test/runtests.py -m apache {posargs:}
+    aws: python test/runtests.py -m aws {posargs:}
+    postgres: python test/runtests.py -m postgres {posargs:}
+    scheduler: python test/runtests.py -m scheduler {posargs:}
+    cdh,hdp: python test/runtests.py -m minicluster {posargs:}
+    gcloud: python test/runtests.py -m gcloud {posargs:}
+    unixsocket: python test/runtests.py -m unixsocket {posargs:}
+    dropbox: python test/runtests.py -m dropbox {posargs:}
+    azureblob: python test/runtests.py -m azureblob {posargs:}
+    core: python test/runtests.py --doctest-modules -m "not minicluster and not gcloud and not postgres and not unixsocket and not contrib and not apache and not aws and not azureblob and not dropbox" {posargs:}
+    ##
     codecov -e TOXENV
     azureblob: {toxinidir}/scripts/ci/stop_azurite.sh
 
 [testenv:visualiser]
 usedevelop = True
 deps =
     mock<2.0
-    nose<2.0
-    unittest2<2.0
     selenium==3.0.2
 passenv = {[testenv]passenv}
 setenv =
     LC_ALL = en_US.utf-8
-    NOSE_EVAL_ATTR=not minicluster and not gcloud and not postgres and not unixsocket
     LUIGI_CONFIG_PATH={toxinidir}/test/testconfig/luigi.cfg
     TEST_VISUALISER=1
 commands =
     python --version
-    nosetests -v --tests=test/visualiser
+    pytest -m "not minicluster and not gcloud and not postgres and not unixsocket" test/visualiser
```

Comment:
why these specific filters here?

Reply:
This filter is counter part of `NOSE_EVAL_ATTER` above. I just replaced it.